tasks.py

```diff
@@ -66,15 +68,28 @@ def download_artifacts(session):
         response.raise_for_status()
         for artifact in response.json()["artifacts"]:
             response = session.get(
-                "{0}artifact/{1}".format(run["url"], artifact["relativePath"])
+                "{0}artifact/{1}".format(run["url"], artifact["relativePath"]),
+                stream=True
+            )
+            assert response.headers["content-length"]
+            print("Downloading {0}".format(artifact["fileName"]))
+            bar = ProgressBar(
+                expected_size=int(response.headers["content-length"]),
+                filled_char="="
             )
+            content = b''
+            for data in response.iter_content(chunk_size=8192):
+                content += data
```

Comment:
`io.BytesIO` is a good solution

On Sat, Nov 7, 2015 at 7:45 PM, Paul Kehrer <notifications@github.com>
wrote:

> In tasks.py
> <https://github.com/pyca/cryptography/pull/2469#discussion_r44219198>:
>
> >              )
> > +            content = b''
> > +            for data in response.iter_content(chunk_size=8192):
> > +                content += data
>
> Yeah I figured it was gonna be bad since it'll copy every time. I
> originally had this as an array and then it only gets concatenated at the
> end, but got lazy since I wanted the length. I'll refactor it quickly to be
> a separate length value.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/2469/files#r44219198>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084